pkg/promise/promise_test.go

```diff
@@ -0,0 +1,28 @@
+package promise
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGo(t *testing.T) {
+	errCh := Go(functionWithError)
+	er := <-errCh
+	require.NotNil(t, er.Error())
+	require.EqualValues(t, "Error Occured", er.Error())
+	t.Log("GO promise with channel value 'Error' executed successfully")
```

Comment:
I didnt implement subtests because there were only two cases that were needed to be tested, or are you suggesting to test multiple scenarios for those two cases ?  Either way I agree both of them are achievable. Please let me know whichever you are referring to.